onnxruntime/python/tools/kernel_explorer/kernels/gemm_test.py

```diff
@@ -33,47 +24,81 @@ def _test_gemm(func, dtype: str, m: int, n: int, k: int, transa=False, transb=Fa
     np.random.seed(0)
     a = (np.random.rand(*a_shape) * 2 - 1).astype(dtype)
     b = (np.random.rand(*b_shape) * 2 - 1).astype(dtype)
+    ref_c = (a.T if transa else a) @ (b.T if transb else b)
 
     my_c = np.zeros((m, n), dtype=dtype)
     dev_a = ke.DeviceArray(a)
     dev_b = ke.DeviceArray(b)
     dev_c = ke.DeviceArray(my_c)
 
-    f = getattr(ke, func)
-
     opa = ke.blas_op.T if transa else ke.blas_op.N
     opb = ke.blas_op.T if transb else ke.blas_op.N
     lda = a_shape[1]
     ldb = b_shape[1]
     alpha = 1.0
     beta = 0.0
-    my_gemm = f(opa, opb, m, n, k, alpha, dev_a, lda, dev_b, ldb, beta, dev_c, n)
-    my_gemm.Run()
-    dev_c.UpdateHostNumpyArray()
+    my_gemm = func(opa, opb, m, n, k, alpha, dev_a, lda, dev_b, ldb, beta, dev_c, n)
 
-    rtol = 1e-3 if dtype == "float16" else 1e-5
-    atol = 1e-3 if dtype == "float16" else 1e-5
+    failures = {}
+    for impl in my_gemm.ListImpls():
+        if not my_gemm.SelectImpl(impl):
+            continue
 
-    print(a, b)
-    ref_c = (a.T if transa else a) @ (b.T if transb else b)
-    np.testing.assert_allclose(ref_c, my_c, rtol=rtol, atol=atol)
+        my_gemm.Run()
+        dev_c.UpdateHostNumpyArray()
+
+        rtol = 1e-3 if dtype == "float16" else 1e-5
+        atol = 1e-3 if dtype == "float16" else 1e-5
+
+        try:
+            np.testing.assert_allclose(ref_c, my_c, rtol=rtol, atol=atol)
+        except Exception as err:
+            failures[impl] = str(err)
 
+    if failures:
+        raise Exception(failures)
 
-dtypes = ["float32", "float16"]
-transabs = product([True, False], repeat=2)
-basic_sizes = product([1, 3, 4, 16, 127, 128, 129, 133, 1024], repeat=3)
+
+def get_basic_cases():
+    dtypes = ["float32", "float16"]
+    transabs = product([True, False], repeat=2)
+    basic_sizes = product([1, 3, 4, 16, 127, 128, 129, 133, 1024], repeat=3)
+    return list(product(dtypes, basic_sizes, transabs))
+
+
+def get_bert_cases():
+    dtypes = ["float32", "float16"]
+    transabs = [(False, False)]
+    bert_sizes = [
+        # m, n, k
+        (384, 64, 384),
```

Comment:
We can profile for both batch size = 1 and 64 cases:
(384*batch_size, 768, 768)
(384*batch_size, 768, 768*3)
(384*batch_size, 768, 768*4)
(384*batch_size, 768*4, 768)

(384*batch_size, 1024, 1024)
(384*batch_size, 1024, 1024*3)
(384*batch_size, 1024, 1024*4)
(384*batch_size, 1024*4, 1024)

Remove (384, 64, 384)?